plugins/network-elements/nuage-vsp/src/com/cloud/agent/api/element/ImplementVspCommand.java

```diff
@@ -0,0 +1,295 @@
+            this._egressFirewallRules = egressFirewallRules;
+            return this;
+        }
+
+        public Builder acsFipUuid(List<String> acsFipUuid) {
+            this._acsFipUuid = acsFipUuid;
+            return this;
+        }
+
+        public Builder egressDefaultPolicy(boolean egressDefaultPolicy) {
+            this._egressDefaultPolicy = egressDefaultPolicy;
+            return this;
+        }
+
+        @Override
+        public ImplementVspCommand build() {
+            return new ImplementVspCommand(_networkId, _networkDomainUuid, _networkUuid, _networkName, _vpcOrSubnetUuid, _isL2Network, _isL3Network, _isVpc, _isShared,
+                    _domainTemplateName, _isFirewallServiceSupported, _dnsServers, _ingressFirewallRules, _egressFirewallRules, _acsFipUuid, _egressDefaultPolicy);
+        }
+    }
+
+    @Override
+    public boolean executeInSequence() {
+        return false;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        ImplementVspCommand that = (ImplementVspCommand) o;
+
+        if (_egressDefaultPolicy != that._egressDefaultPolicy) return false;
+        if (_isFirewallServiceSupported != that._isFirewallServiceSupported) return false;
+        if (_isL2Network != that._isL2Network) return false;
+        if (_isL3Network != that._isL3Network) return false;
+        if (_isShared != that._isShared) return false;
+        if (_isVpc != that._isVpc) return false;
+        if (_networkId != that._networkId) return false;
+        if (_acsFipUuid != null ? !_acsFipUuid.equals(that._acsFipUuid) : that._acsFipUuid != null) return false;
+        if (_dnsServers != null ? !_dnsServers.equals(that._dnsServers) : that._dnsServers != null) return false;
+        if (_domainTemplateName != null ? !_domainTemplateName.equals(that._domainTemplateName) : that._domainTemplateName != null)
+            return false;
+        if (_egressFirewallRules != null ? !_egressFirewallRules.equals(that._egressFirewallRules) : that._egressFirewallRules != null)
+            return false;
+        if (_ingressFirewallRules != null ? !_ingressFirewallRules.equals(that._ingressFirewallRules) : that._ingressFirewallRules != null)
+            return false;
+        if (_networkDomainUuid != null ? !_networkDomainUuid.equals(that._networkDomainUuid) : that._networkDomainUuid != null)
+            return false;
+        if (_networkName != null ? !_networkName.equals(that._networkName) : that._networkName != null) return false;
+        if (_networkUuid != null ? !_networkUuid.equals(that._networkUuid) : that._networkUuid != null) return false;
+        if (_vpcOrSubnetUuid != null ? !_vpcOrSubnetUuid.equals(that._vpcOrSubnetUuid) : that._vpcOrSubnetUuid != null)
+            return false;
+
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
This ``hashCode`` implementation should include attributes from the ``Command`` super class.